SofascoreAcademy/SofascoreAcademy/Controllers/MainViewController.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  MainViewController.swift
+//  SofascoreAcademy
+//
+//  Created by Matija Pavicic on 02.04.2024..
+//
+
+import Foundation
+import SnapKit
+import SofaAcademic
+import UIKit
+
+class MainViewController: UIViewController, BaseViewProtocol, TabItemDelegateProtocol {
+    
+    private let safeAreaCoverView: SafeAreaCoverView = .init()
+    private let mainHeaderView: MainHeaderView = .init()
+    private let eventsViewController: EventsViewController = .init()
+    private let settingsViewController: SettingsViewController = .init()
+    private let containerView: BaseView = .init()
+    private let tabView: TabView = .init()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tabView.delegate = self
+        
+        addViews()
+        setupConstraints()
+        styleViews()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        navigationController?.isNavigationBarHidden = true
+        
+        eventsViewController.dataIndex(UserDefaults.standard.integer(forKey: "tabIndex"))
+    }
+    
+    func addViews() {
+        view.addSubview(safeAreaCoverView)
+        view.addSubview(mainHeaderView)
+        view.addSubview(tabView)
+        view.addSubview(containerView)
+        add(eventsViewController)
+    }
+    
+    func setupConstraints() {
+        safeAreaCoverView.snp.makeConstraints {
+            $0.bottom.equalTo(view.safeAreaLayoutGuide.snp.top)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(60)
+        }
+        mainHeaderView.snp.makeConstraints {
+            $0.top.equalTo(safeAreaCoverView.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(48)
+        }
+        tabView.snp.makeConstraints {
+            $0.top.equalTo(mainHeaderView.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(48)
+        }
+        containerView.snp.makeConstraints {
+            $0.top.equalTo(tabView.snp.bottom)
+            $0.leading.trailing.bottom.equalToSuperview()
+        }
+    }
+    
+    func styleViews() {
+        mainHeaderView.settingButtonTapHandler = {
+            self.showSettings()
+        }
+    }
+    
+    func tabView(_ tabView: TabView, didSelectTabAtIndex index: Int) {
+        
+        let eventsViewController = EventsViewController()
```

Comment:
Trenutno nemaš animaciju kada se u contraineru mijenja ViewController. Zapravo bi trebao imati 3 UIViewControllera, za svaki sport i njih izmjenjivati na ovaj način i staviti animaciju za tranziciju.